### ISSUE TITLE
Add ability to use QA server in tCore

### ISSUE THREAD

**Issue by PhotoNomad0** [unfoldingWord/translationCore#7449]
### User Story

As a [type of user], I want [some goal] so that [some reason].

### Features / Specifications

- [ ]
- [ ]
- [ ]

### Definition of Done

- [ ]
- [ ]
- [ ]

**Comment by PhotoNomad0** [comment]
**Notes:**

In GWE here are the paths:
```
export const BASE_URL = 'https://git.door43.org'
export const QA_BASE_URL = 'https://qa.door43.org'
```

**Comment by PhotoNomad0** [comment]
Notes:
- a toggle for QA server in tCore is a bit involved - each tool is independent, and tc-source-content-downloader is independent.
- qa server needs to work for:
  - login
  - resource downloads
  - catalog next queries
  - project uploads/downloads

**Comment by PhotoNomad0** [comment]
@birchamp @elsylambert - We will need an easy way for user to remove downloaded QA content so that it does not impact operation of the regular tCore.  Have a few options:
1. use a different resource folder for QA (e.g. resource-QA).  Will consume more disk space, but no chance of contamination of the resources in the regular app. Also add another button to bottom of resource download dialog to "Delete All QA resources" to free up disk space once they are done testing.
2. add way of resetting resources to just the bundled resources
3. flag qa downloaded resources in manifest so they can be removed easily.  Also add another button to bottom of resource download dialog to "Delete All QA resources"

It seems that option one is the safest.

**Comment by birchamp** [comment]
@PhotoNomad0 it also seems like option 1 is the simplest to develop.  Since this isn't intended for every user, I think the disk space issue is a good compromise. 

Also I think it should be obvious in the titlebar or a UI color or something that the user is on the QA server.

**Comment by elsylambert** [comment]
@PhotoNomad0 Yes Option 1, it looks similar to the way we have pre-released Resources deleted. And downloads in a separate folder like we have for USFM URLs
<img width="855" alt="Screenshot 2022-11-11 at 8 55 40 AM" src="https://user-images.githubusercontent.com/51806180/201354909-84a660c6-5342-4add-9234-4c419101b13a.png">



Since we are adding a way to toggle between QA and Prod servers, Would it make sense to add the option to toggle on the user login page.
<img width="1635" alt="Screenshot 2022-11-11 at 8 52 07 AM" src="https://user-images.githubusercontent.com/51806180/201354408-59a9b094-493a-4c7e-b991-ae4e3abbb1a2.png">
<img width="1635" alt="Screenshot 2022-11-11 at 8 50 42 AM" src="https://user-images.githubusercontent.com/51806180/201354453-05f81710-8fdd-46c0-89d4-f03d53af9067.png">

**Comment by PhotoNomad0** [comment]
Notes:
- when in QA mode - tC will store data in `~/translationCore-QA`
- the safest way to enable QA mode is at startup.  Trying this to enable QA mode before startup: 
  - on Mac to enable QA mode before startup, in terminal window do `echo "https://qa.door43.org" > ~/translationCore/QA_MODE` to disable `rm ~/translationCore/QA_MODE`
  - on Windows to enable QA mode before startup, in command Prompt do `echo https://qa.door43.org > %HOMEDRIVE%%HOMEPATH%\translationCore\QA_MODE` to disable `del %HOMEDRIVE%%HOMEPATH%\translationCore\QA_MODE`

- ~~each time you switch in and out of QA mode, you will have to log out and log in again.~~ <= Changed to save user credentials by server in QA mode.  So the first time you switch servers, you will have to log in again.

**Comment by PhotoNomad0** [comment]
This is how we are showing we are in QA mode:

<img width="1087" alt="Screenshot 2022-11-15 at 6 59 19 AM" src="https://user-images.githubusercontent.com/14238574/201914966-1142db78-940f-4740-9d29-4594af056f83.png">

**Comment by PhotoNomad0** [comment]
@elsylambert - Have a test build for you to look at: `translationCore 3.4.0-QA (da05ef4)`

See Notes above on how to enable QA mode.

**Comment by PhotoNomad0** [comment]
USFM import not preserving alignments - looks like folder naming problem.

**Comment by PhotoNomad0** [comment]
@elsylambert This is in translationCore 3.4.0-QA (https://github.com/unfoldingWord/translationCore/commit/d482d0937fc31e77fcb216e3e08d95c64a071318).

**Comment by elsylambert** [comment]
Looks good in translationCore 3.4.0-QA (d482d09).
